### ISSUE TITLE
Questions of some implementation details

### ISSUE THREAD

**Issue by Victor0118** [NTMC-Community/MatchZoo#442]
1. In the [MVLSTM paper](https://arxiv.org/pdf/1511.08277.pdf), the author used Cosine/Bilinear/Tensor function to model interactions while in Matchzoo implementation (https://github.com/NTMC-Community/MatchZoo/blob/master/matchzoo/models/mvlstm.py#L53), you use Dot product if I understand correctly.  Would you consider adding them into Matchzoo?

2. In the [aNMM paper](http://maroo.cs.umass.edu/pub/web/getpdf.php?id=1240), the author used two configurations: aNMM-1 and aNMM-2. The basic difference is whether the hidden layer shares weight or not. According to my understanding of Matchzoo implementation, you only implemented aNMM-1, right? Did you try both and keep the better structure?  

3. The same question to Matchpyramid: in the original paper, they compare cosine and doc product as the intersection function. In the Matchzoo implementation, you use [cosine](https://github.com/NTMC-Community/MatchZoo/blob/master/matchzoo/models/matchpyramid.py#L53).

Thanks!

**Comment by faneshion** [comment]
Yeah. The models you have mentioned above have different implementations since there are different choices in some specific layers. In the previous version(i.e., the master branch), we only selected the best choice in our data. Right now, we are working on the 2.0 version, we will make it as more flexible as possible. By the way, we are looking forward to more developers to join us to make this tool more better.

**Comment by faneshion** [comment]
Yeah. The models you have mentioned above have different implementations since there are different choices in some specific layers. In the previous version(i.e., the master branch), we only selected the best choice in our data. Right now, we are working on the 2.0 version, we will make it as more flexible as possible. By the way, we are looking forward to more developers to join us to make this tool better.

**Comment by yangliuy** [comment]
@Victor0118 Thank you for your question! For the question on aNMM, the main difference between aNMM2 and aNMM1 is that there is one additional hidden layer in aNMM2 which could enable us to use multiple sets of value-shared weights for each QA matching vector. In the current implementation, you can set num_layers to change the number of layers in this feed forward neural networks. As shown in the paper, the performance of aNMM2 is not necessarily better then aNMM1. It really depends on the size and distribution of your data and labels. More complex models are more likely to overfit the data.

**Comment by pl8787** [comment]
Please see (MatchLayer)[https://github.com/NTMC-Community/MatchZoo/blob/master/matchzoo/layers/Match.py] and (MatchTensorLayer)[https://github.com/NTMC-Community/MatchZoo/blob/master/matchzoo/layers/MatchTensor.py].

**Comment by Victor0118** [comment]
@pl8787 @yangliuy @faneshion Thanks for your reply! I think usually we choose a structure that performs well in one test set but we cannot ganrentee this advantage generalize to other domains and datasets. The best option should be keeping all options and let the users do the configuration.

 I'd like to help with the version 2.0 since I will work with the toolkit for a while. Personally I think Matchzoo is great but definitely need improvement.

**Comment by faneshion** [comment]
That would be nice if you can join us. Right now,  we have a large number of models to be implemented under the 2.0 framework. You can refer to [this file](https://github.com/NTMC-Community/MatchZoo/blob/2.0/CONTRIBUTING.md) to see how to contribute. You can familiar with the whole process first.

**Comment by Victor0118** [comment]
@faneshion Thanks! I'll pick a model to implement.
